src/FormTextField/FormTextField.js

```diff
@@ -0,0 +1,37 @@
+import React from "react";
+// import PropTypes from "prop-types";
+// import { dcnb } from "cnbuilder";
+import { Controller } from "react-hook-form";
+import { TextField } from "../TextField/TextField";
+// import {} from "./FormTextField.levers";
+
+/**
+ * Text Input  Component
+ *
+ */
+export const FormTextField = ({ name, control, label, error }, props) => {
+  // Defaults & Variables.
+  // ---------------------------------------------------------------------------
+  const placeholder = "";
+
+  return (
+    <Controller
+      name={name}
+      control={control}
+      render={({ onChange }) => (
+        <TextField
+          label={label}
+          onChange={onChange}
+          errorText={error ? "Error!" : null}
+          error={error}
+          {...props}
+        />
+      )}
+    />
+  );
+};
```

Comment:
Out of curiosity, is there a reason we need to wrap this with a `Controller` rather than just [registering the component](https://react-hook-form.com/api/useform/register)? If we can register rather than control our form components will be much more performant.

Reply:
Ooo great question @mattanglin! 🤔 Based on my understanding of the react-hook-forms (RHF), when using a styled component library with RHF, we want to use the [Controller wrapper component](https://react-hook-form.com/api/usecontroller/controller) since it provides and easier way to handle the state of the form data while passing down additional props. However, I see that `register` also contains similar 

I can definitely start another branch and open up a PR for experimenting with `register` and see how that combination would work as well 😄 

It seems to comes down to how we want to build out and pass information down to our form element components in Decanter-React along with how we want our form data to be handled: React state vs DOM?

Reply:
Both of the methods will provide the same result from a functionality perspective, but using `register` is significantly more performant as it allows RHF to work with form inputs without having to completely control them. This means less unnecessary renders for child components. The only caveat to this is that the form inputs need to accept a `ref` and standard input event handlers. The RHF `Controller` is provided as a workaround around for component libs that choose to eschew the input attribute standards and expose event handlers that directly pass the input's value as handler params instead of the native element target event (like material).
For something like a simple text input this means the difference between re-rendering every form input on every keypress vs just rendering when major global form state changes. Try creating a registered and controlled text input and logging from each of them and you should see a stark difference in render counts.
I would argue that since we have control over how our form inputs are implemented we should forward the input ref and handle input events naturally and only use `Control` when absolutely necessary. Otherwise, we lose the [key benefits](https://react-hook-form.com/faqs#PerformanceofReactHookForm) that RHF offers as a form library.

Reply:
Thank you for the detailed explanation @mattanglin!!! 😄  All good to know and helps bring a lot of clarity while I test out and put together an example of the register implementation.